src/main.py

```diff
@@ -44,43 +58,103 @@ def on_exit_btn(self) -> None:
         App.get_running_app().stop()
 
     async def app_func(self):
-        async def run_wrapper() -> None:
+        async def run_wrapper():
             # we don't actually need to set asyncio as the lib because it is
             # the default, but it doesn't hurt to be explicit
             await self.async_run(async_lib="asyncio")
-            for task in self.async_tasks:
+            for task in self.tasks:
                 task.cancel()
 
-        # Placeholder task
-        self.async_tasks.append(asyncio.ensure_future(self.template_function()))
+        # configure the camera client
+        config = ClientConfig(address=self.address, port=self.port)
+        client = OakCameraClient(config)
+
+        # Stream camera frames
+        self.tasks.append(asyncio.ensure_future(self.stream_camera(client)))
```

Comment:
Tasks is not really necessary to be declared as futures.

something like this will work as well
```python
return await asyncio.gather(run_wrapper(), self.stream_camera(client))
```

Reply:
We should remove from app-template then. I will open PR's to change this change in:
1. app-template
2. camera-streamer
3. virtual-joystick

Reply:
I think we keep the list of tasks as a placeholder though, to keep it is simpler for users to expand custom apps to multiple tasks